config/config.go

```diff
@@ -1545,94 +1547,125 @@ func migrateConfig(configFile, targetDir string) (string, error) {
 	return target, nil
 }
 
-// ReadConfig verifies and checks for encryption and verifies the unencrypted
-// file contains JSON.
-// Prompts for decryption key, if target file is encrypted
-func (c *Config) ReadConfig(configPath string, dryrun bool) error {
+// ReadConfigFromFile reads the configuration from the given file
+// if target file is encrypted, prompts for encryption key
+// Also - if not in dryrun mode - it checks if the configuration needs to be encrypted
+// and stores the file as encrypted, if necessary (prompting for enryption key)
+func (c *Config) ReadConfigFromFile(configPath string, dryrun bool) error {
 	defaultPath, _, err := GetFilePath(configPath)
 	if err != nil {
 		return err
 	}
-
-	fileData, err := ioutil.ReadFile(defaultPath)
+	confFile, err := os.Open(defaultPath)
 	if err != nil {
 		return err
 	}
+	defer confFile.Close()
+	result, wasEncrypted, err := ReadConfig(confFile, func() ([]byte, error) { return PromptForConfigKey(false) })
+	if err != nil {
+		return fmt.Errorf("error reading config %w", err)
+	}
+	// Override values in the current config
+	*c = *result
 
-	if !ConfirmECS(fileData) {
-		err = json.Unmarshal(fileData, c)
-		if err != nil {
-			return err
-		}
-
-		if c.EncryptConfig == fileEncryptionDisabled {
-			return nil
-		}
+	if dryrun || wasEncrypted || (!wasEncrypted && c.EncryptConfig == fileEncryptionDisabled) {
+		return nil
+	}
 
-		if c.EncryptConfig == fileEncryptionPrompt {
-			confirm, err := promptForConfigEncryption()
-			if err == nil {
-				if confirm {
-					c.EncryptConfig = fileEncryptionEnabled
-					return c.SaveConfig(defaultPath, dryrun)
-				}
+	if c.EncryptConfig == fileEncryptionPrompt {
+		confirm, err := promptForConfigEncryption()
```

Comment:
Can we please check this error and return? Or maybe drop the error return altogether.

Reply:
This was the behaviour before - it (or the lack of error) is already checked, I guess the returned error is ignored (not propagated), because even if the user prompt for encryption fails (e.g. running in some non-interactive mode), we've already loaded the configuration and it can be used, but yes I agree the user is entitled to at least log message if they've requested for encryption prompt (and at this stage have not confirmed plain text or encryption option), but it failed. 
A similar case is a few lines below, after they've confirmed plaintext option, but the choice cannot be saved (e.g. file permissions problems)

"Dropping the error" option I don't like as we'd want to repeat the prompt next time instead of defaulting to not encrypted.

Reply:
What you have done is reasonable. Thank you very much!